creator/data_templates/mutations/template_version.py

```diff
@@ -0,0 +1,259 @@
+import graphene
+from graphql import GraphQLError
+from graphql_relay import from_global_id
+from django.db import transaction
+from django.conf import settings
+
+from creator.studies.models import Study
+from creator.data_templates.models import (
+    DataTemplate,
+    TemplateVersion,
+)
+from creator.data_templates.nodes.template_version import TemplateVersionNode
+
+
+def check_studies(study_node_ids):
+    """
+    Check if studies exist and return the Study objects if all studies exist
+
+    Helper function called in create/update template version mutations
+    """
+    # Convert graphql relay ids to primary keys
+    study_ids = []
+    for sid in study_node_ids:
+        _, study_id = from_global_id(sid)
+        study_ids.append(study_id)
+
+    # Check all studies exist
+    studies = Study.objects.filter(pk__in=study_ids).only("kf_id").all()
+    if len(studies) != len(study_ids):
+        raise GraphQLError(
+            "Failed to create/update template version because one or more "
+            "studies in the input do not exist. "
+        )
+    return studies
+
+
+class CreateTemplateVersionInput(graphene.InputObjectType):
+    """Parameters used when creating a new template_version"""
+
+    description = graphene.String(
+        required=True, description="The description of the template_version"
+    )
+    field_definitions = graphene.JSONString(
+        required=True, description="The field definitions for this template"
+    )
+    data_template = graphene.ID(
+        required=True,
+        description="The data_template that this template_version belongs to",
+    )
+    studies = graphene.List(
+        graphene.ID,
+        description="The studies this template version should be assigned to",
+    )
+
+
+class UpdateTemplateVersionInput(graphene.InputObjectType):
+    """Parameters used when updating an existing template_version"""
+
+    description = graphene.String(
+        description="The description of the template_version"
+    )
+    field_definitions = graphene.JSONString(
+        description="The field definitions for this template"
+    )
+    studies = graphene.List(
```

Comment:
this replaces the list of studies each time? to add a study you would pass all current studies plus the new ones?

Reply:
Yea this would replace the list each time. I think this would _probably_ only be useful when you create a template version. I doubt there will be many times where you update a template version and replace the studies it's linked to.

For those reasons, we will also have mutations to link/unlink template versions to/from studies: https://github.com/kids-first/kf-api-study-creator/issues/676